### PR TITLE
Intgrtns 204 implement the  config validate function

### PR DESCRIPTION
**Pull request by Uditha-05**
Added validation function(openprovider_config_validate) for user-given registrar configuration.
Validated the given configurations with the given environment(Sandbox or Production).
Reduced too many "generateAuthTokenRequest" requests invoking.

### CODE REVIEW

File: modules/registrars/openprovider/openprovider.php [hunk 1]
```diff
@@ -329,6 +330,50 @@ function openprovider_ResendIRTPVerificationEmail(array $params)
     return openprovider_registrar_launch_decorator('resendIRTPVerificationEmail', $params);
 }
 
+
+function openprovider_config_validate($params)
+{
+    $username = $params['Username'];
+    $password = $params['Password'];
+    $test = $params['test_mode'];
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/openprovider.php:338]: better to use the variable name as it's (test_mode)

File: modules/registrars/openprovider/openprovider.php [hunk 2]
```diff
@@ -329,6 +330,50 @@ function openprovider_ResendIRTPVerificationEmail(array $params)
...
+    $ip = $_SERVER['SERVER_ADDR'];
+
+    $valid = false;
+
+    if ($test == 'on') {
+        $env = 'Sandbox';
+        $url = 'http://api.sandbox.openprovider.nl:8480/v1beta/auth/login';
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/openprovider.php:345]: make these values as constant in the file atleast, or if we have any config files that's better

File: modules/registrars/openprovider/openprovider.php [hunk 3]
```diff
@@ -329,6 +330,50 @@ function openprovider_ResendIRTPVerificationEmail(array $params)
...
+    } else {
+        $env = 'Production';
+        $url = 'https://api.openprovider.eu/v1beta/auth/login';
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/openprovider.php:348]: this too

File: modules/registrars/openprovider/Controllers/System/ConfigController.php [hunk 4]
```diff
@@ -179,8 +179,8 @@ protected function generateLoginError(array $configarray, $error)
     protected function checkCredentials($configarray, $params)
     {
         $differentHost = $params['test_mode'] == 'on' ?
-            Configuration::get('api_url') :
-            Configuration::get('api_url_cte');
+            Configuration::get('api_url_cte') :
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/Controllers/System/ConfigController.php:182]: strange, this worked for long time
please make sure

File: modules/registrars/openprovider/openprovider.php (hunk 1 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/openprovider.php]: This change has been added

File: modules/registrars/openprovider/openprovider.php (hunk 2 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/openprovider.php]: These changes have been added

File: modules/registrars/openprovider/openprovider.php (hunk 3 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/openprovider.php]: These changes have been added

File: modules/registrars/openprovider/Controllers/System/ConfigController.php (hunk 4 above)
Comment (Uditha-05) [reply modules/registrars/openprovider/Controllers/System/ConfigController.php:182]: Confirmed! `Configuration::get('api_url_cte')` is for Sandbox/Customer Test Environment

File: modules/registrars/openprovider/openprovider.php [hunk 5]
```diff
@@ -329,6 +330,49 @@ function openprovider_ResendIRTPVerificationEmail(array $params)
...
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($curl, CURLOPT_HTTPHEADER, array('Content-Type:application/json'));
+    curl_setopt($curl, CURLOPT_POST, true);
+    curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
+    curl_setopt($curl, CURLOPT_POSTFIELDS, $encodedData);
+    $result = curl_exec($curl);
+    curl_close($curl);
+    $response = json_decode($result);
+    if ($response->data->token && $response->data->reseller_id) {
+        $valid = true;
```
Comment (sampikaipakan) [inline modules/registrars/openprovider/openprovider.php:369]: you can remove this $valid variable and check directly on these values

`if (!$response->data->token || ! $response->data->reseller_id) {
  throw Error...
  }

Comment (Uditha-05) [reply modules/registrars/openprovider/openprovider.php]: This change has been added

**Review by sampikaipakan — APPROVED** [review]

**Pull request merged**